src/util/datablock/datablock.c

```diff
@@ -180,3 +180,28 @@ void DataBlock_Free(DataBlock *dataBlock) {
 	assert(pthread_mutex_destroy(&dataBlock->mutex) == 0);
 	rm_free(dataBlock);
 }
+
+/* --------- Out of Order interface implementation --------*/
+
+void *DataBlock_AllocateItemOutOfOrder(DataBlock *dataBlock, uint64_t idx) {
+	DataBlock_Accommodate(dataBlock, idx);
+	dataBlock->itemCount++;
+	Block *block = GET_ITEM_BLOCK(dataBlock, idx);
+	idx = ITEM_POSITION_WITHIN_BLOCK(idx);
+	DataBlockItemHeader *item_header = (DataBlockItemHeader *)block->data + (idx * block->itemSize);
```

Comment:
getting the header is becoming a common operation, please introduce a function which given a block and an item index retrieves the item header.